[docs-v2]: transfer pipeline release checklist (#3745)
nf-core/website: sites/docs/src/content/docs/developing/pipelines/release-procedure.md

subtitle: A step-by-step guide for releasing a nf-core pipeline
markdownPlugin: checklist
---

## Before you release

:::warning
If this is your first release, follow the [first release guide](/docs/contributing/contribute-new-pipelines#make-the-first-release) for extra review requirements
:::

- [ ] Check on the issues tab of GitHub repository that there are no outstanding bug reports that should be resolved.
- [ ] If you have a [release milestone](https://help.github.com/en/github/managing-your-work-on-github/about-milestones), check all planned functionality and improvements are addressed.
  - [ ] Move any postposed issues to the next milestone
- [ ] Check the pipeline follows [nf-core guidelines](/docs/specifications/pipelines/overview) and [nf-core pipeline release review guidelines](/docs/contributing/reviewing-pull-requests/review_checklists/pipeline-release).
- [ ] Check that all continuous-integration tests on the `dev` branch pass
  - [ ] Check the output of linting for warnings and address all that you can resolve.

    ```bash
    nf-core pipelines lint
    ```

  - [ ] If possible, update any software dependencies that are reported to be out of date.

> 

Author: ```suggestion
```

---
title: Release procedure
subtitle: A step-by-step guide for releasing a nf-core pipeline
markdownPlugin: checklist
weight: 10
---

## Before you release

:::warning

> If this is your first release, make sure to follow the [first release tutorial](/docs/contributing/contribute-new-pipelines#make-the-first-release) for extra review requirements!

Reviewer: ```suggestion
If this is your first release, follow the [first release guide](/docs/contributing/contribute-new-pipelines#make-the-first-release) for extra review requirements
```

:::

- [ ] Check on the issues tab of GitHub repository that there are no outstanding bug reports that should be resolved.
- [ ] If you have a [release milestone](https://help.github.com/en/github/managing-your-work-on-github/about-milestones), check all planned functionality and improvements are addressed.

>   - It's fine to decide that some things should be postponed until the next release. Move them on to the next milestone

Reviewer: ```suggestion
  - [ ] Move any postposed issues to the next milestone
```

- [ ] Check the pipeline follows [nf-core guidelines](/docs/specifications/pipelines/overview) and [nf-core pipeline release review guidelines](/docs/contributing/reviewing-pull-requests/review_checklists/pipeline-release).
  - This is what reviewers will be checking against.
- [ ] Check that all continuous-integration tests on the `dev` branch are passing.
  - [ ] Check the output of linting for warnings and address all that you can resolve.

    ```bash
    nf-core pipelines lint
    ```

  - [ ] If possible, update any software dependencies that are reported to be out of date.

- [ ] Check that the full-size tests have successfully completed within the `dev` branch.
- [ ] Review the `CHANGELOG.md` to ensure it includes everything that has been added/fixed in this release.

>   - We recommend you also add the GitHub handle of the main contributors of each CHANGELOG entry (author, and significant reviewers etc.).

Reviewer: ```suggestion
  - Add the GitHub handle of the main contributors of each CHANGELOG entry (author, significant reviewers, etc.).
```
